azure-pipelines/templates/stages/release.yml

```diff
@@ -1,58 +1,77 @@
-                  scriptType: ps
-                  scriptLocation: inlineScript
-                  inlineScript: |
-                    az storage blob upload-batch --account-name ${{ stage.templateContext.storage }} -d "packages" --destination-path "${{ target }}/fiskaltrust.Launcher/" -s "$(Pipeline.Workspace)/${{ parameters.artifact }}/package-${{ target }}/" --pattern "*.zip*"
\ No newline at end of file
+  - ${{ each stage in parameters.stages }}:
+      - ${{ each pair in stage }}:
+          ${{ pair.key }}: ${{ pair.value }}
+        
+        displayName: Release ${{ stage.templateContext.environment }}
+        
+        ${{ if eq(stage.templateContext.environment, 'production') }}:
+          condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/v')
+        
+        jobs:
+          - deployment: Release
+            
+            pool:
+              vmImage: windows-latest
+            
+            environment: ${{ stage.templateContext.serviceConnection }}
+            
+            strategy:
+              runOnce:
+                deploy:
+                  steps:
+
+                    - ${{ each target in parameters.targets }}:
+                        - download: ${{ parameters.artifact }}
+                          artifact: package-${{ target }}
+                          displayName: "[${{ target }}] Download artifact"
+
+                    - pwsh: get-childitem "$(Pipeline.Workspace)/${{ parameters.artifact }}/package-*/*" | foreach { rename-item $_ $_.Name.Replace("fiskaltrust.Launcher-", "") }
+                      displayName: Rename artifacts
+
+                    - ${{ each target in parameters.targets }}:
+                        - task: AzureCLI@2
+                          displayName: "[${{ target }}] Publish to blob storage"
+                          inputs:
+                            azureSubscription: ${{ stage.templateContext.serviceConnection }}
+                            scriptType: ps
+                            scriptLocation: inlineScript
+                            inlineScript: |
+                              az storage blob upload-batch --account-name ${{ stage.templateContext.storage }} -d "packages" --destination-path "${{ target }}/fiskaltrust.Launcher/" -s "$(Pipeline.Workspace)/${{ parameters.artifact }}/package-${{ target }}/" --pattern "*.zip*"
+
+                    - ${{ each target in parameters.targets }}:
+                        - task: CopyFiles@2
+                          displayName: "[${{ target }}] Include scripts in the package"
+                          inputs:
+                            SourceFolder: "scripts"
+                            Contents: "**"
+                            TargetFolder: "$(Pipeline.Workspace)/${{ parameters.artifact }}/package-${{ target }}"
+                            Overwrite: true
+
+                    - ${{ each target in parameters.targets }}:
+                        - task: ArchiveFiles@2
```

Comment:
We'll need to make sure this task is preserving the executable permissions on linux 🥲 otherwise the linux scripts will not be able to be executed

Reply:
Yes, you were right. Good point. I've changed it like this: for each  operating system, check whether it is a variant of Linux or macOS. If yes, sowe use a bash script to create a zip archive, which preserves the file permissions (so the scripts will still be executable when extracted). For Windows, we use the standard ArchiveFiles task provided by Azure Devops :)